dali/python/nvidia/dali/external_source.py

```diff
@@ -27,18 +27,28 @@
 )
 
 
+def _get_shape(data):
+    if isinstance(data, (_tensors.TensorCPU, _tensors.TensorGPU)):
+        if callable(data.shape):
+            return data.shape()
+        else:
+            return data.shape
```

Comment:
Won't this change behaviour? Are there any types that can represent tensor, with a `shape()` as a callable method? Does having `__cuda_array_interface__` imply `__array_interface__`?